onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -434,11 +436,22 @@ def __init__(
     def _create_inference_session(self, providers, provider_options, disabled_optimizers=None):
         available_providers = C.get_available_providers()
 
-        # Tensorrt can fall back to CUDA. All others fall back to CPU.
+        # Tensorrt can fall back to CUDA if it's explicitly assigned. All others fall back to CPU.
         if "TensorrtExecutionProvider" in available_providers:
-            self._fallback_providers = ["CUDAExecutionProvider", "CPUExecutionProvider"]
+            if any(
```

Comment:
What if user specifies providers = ["TensorrtExecutionProvider", "CPUExecutionProvider"] ?
it seems it will have no fallback provider.

Reply:
I think it will go to [the else case in the end](https://github.com/microsoft/onnxruntime/blob/d5fb0ce42843df13b9daf12fe7ca338b53d1b000/onnxruntime/python/onnxruntime_inference_collection.py#L455) that assign CPU as fallback EP (which will be the same as user specifies providers = ["TensorrtExecutionProvider"])

Reply:
sync up offline. 
will have another fix.

Reply:
else conditions have been fixed and tested. Thanks!